build.csx

```diff
@@ -0,0 +1,225 @@
+
+targets.Add("logsDirectory", () => Directory.CreateDirectory(logsDirectory));
+
+targets.Add("testsDirectory", () => Directory.CreateDirectory(testsDirectory));
+
+targets.Add("build", DependsOn("clean", "outputDirectory", "restore"), () => RunMsBuild("Build"));
+
+targets.Add("clean", DependsOn("logsDirectory"), () => RunMsBuild("Clean"));
+
+targets.Add(
+    "restore",
+    () =>
+    {
+        Cmd(nuget, $"restore {solution} -PackagesDirectory {packagesDir}");
+        foreach (var projectDir in netstdUnitTestDirectories.Concat(netstdIntegrationTestDirectories).Concat(netstdSpecDirectories))
+        {
+            Cmd("dotnet", $"restore {projectDir}");
+        }
+    });
+
+targets.Add(
+    "unit",
+    DependsOn("build", "testsDirectory"),
+    () =>
+    {
+        RunTests(unitTests);
+        RunDotNetTests(netstdUnitTestDirectories);
+    });
+
+targets.Add(
+    "integ",
+    DependsOn("build", "testsDirectory"),
+    () =>
+    {
+        RunTests(integrationTests);
+        RunDotNetTests(netstdIntegrationTestDirectories);
+    });
+
+targets.Add(
+    "spec",
+    DependsOn("build", "testsDirectory"),
+    () =>
+    {
+        RunTests(specs);
+        RunDotNetTests(netstdSpecDirectories);
+    });
+
+targets.Add(
+    "approve",
+    DependsOn("build", "testsDirectory"),
+    () => RunTests(approvalTests));
+
+targets.Add(
+    "pack",
+    DependsOn("build", "outputDirectory"),
+    () =>
+    {
+        var version = GetVersion(assemblyInfo);
+        foreach (var nuspec in nuspecs)
+        {
+            Cmd(nuget, $"pack {nuspec} -Version {version} -OutputDirectory {outputDirectory} -NoPackageAnalysis");
+        }
+    });
+
+targets.Add(
+    "gitlink",
+    DependsOn("build"), () => Cmd(gitlink, $". -f {solution} -u {repoUrl} -include " + string.Join(",", gitlinks)));
+
+Run(Args, targets);
+
+// helpers
+public static void Cmd(string fileName, string args)
+{
+    using (var process = new Process())
+    {
+        process.StartInfo = new ProcessStartInfo { FileName = $"\"{fileName}\"", Arguments = args, UseShellExecute = false, };
+        Console.WriteLine($"Running '{process.StartInfo.FileName} {process.StartInfo.Arguments}'...");
+        process.Start();
+        process.WaitForExit();
+        if (process.ExitCode != 0)
+        {
+            throw new InvalidOperationException($"The command exited with code {process.ExitCode}.");
+        }
+    }
+}
+
+public static void RunMsBuild(string target)
+{
+    var packagesDirOption = string.IsNullOrEmpty(packagesDir) ? "" : $"/p:NuGetPackagesDirectory={packagesDir}";
+    Cmd(
+        msBuild,
+        $"{solution} /target:{target} /p:configuration=Release /nr:false /verbosity:minimal /nologo /fl /flp:LogFile=artifacts/logs/{target}.log;Verbosity=Detailed;PerformanceSummary {packagesDirOption}");
+}
+
+public static void RunDotNet(string workingDirectory, string command, string args="")

```

Comment:
This is an almost complete replica of `Cmd`. I wonder if it would be better to have an overload for `Cmd` which takes a working directory and make the message conditional on whether it is `null` or not?